java/ql/src/Security/CWE/CWE-346/UnvalidatedCors.ql

```diff
@@ -30,9 +30,14 @@ private predicate setsAllowCredentials(MethodAccess header) {
 
 class CorsProbableCheckAccess extends MethodAccess {
   CorsProbableCheckAccess() {
-    getMethod().getName() = ["contains", "equals"] and
-    getMethod().getDeclaringType().getQualifiedName() =
-      ["java.util.List<String>", "java.util.ArrayList<String>", "java.lang.String"]
+    getMethod().hasName("contains") and
+    getMethod()
+        .getDeclaringType()
+        .getASourceSupertype*()
+        .hasQualifiedName("java.util", "Collection")
```

Comment:
I think you could replace this with `CollectionType`.

Reply:
@intrigus-lgtm will do 👍